### PR TITLE
Pythonise reg tests

### PR DESCRIPTION
**Pull request by wdednam**
Example of port to python of tricky TCL script, canvasContext.sh. Will continue in this way if this attempt is acceptable.

- - -
This change is [Reviewable](https://reviewable.io/reviews/highperformancecoder/minsky/520)

### CODE REVIEW

**Review by highperformancecoder — COMMENTED** [review]
The original test was about testing that the context menu posts the correct menu. Being a GUI function, this now needs to be tested in the Javascript environment, so this actually needs to be done as a jest test. I'm still not completely au fait with doing jest tests, and the Javascript code is not written in a test-friendly manner, which is one reason I had looked at this test before for pythonising, and ended up "punting it down the road".

The python code you write looks good, and it is probably valuable in what it tests (would really need to see what lines it covers), so I'd be happy to approve provided we create a ticket about creating an actual context menu test in jest.

**Comment by wdednam** [comment]
I approved the review at Reviewable but it still says it needs to be reviewed, I guess by you?

**Review by highperformancecoder — APPROVED** [review]

**Comment by highperformancecoder** [comment]
Reviewable is still saying there is one unresolved discussion, but I can't see what that is, so I'm pressing the button.

File: .gitignore [hunk 1]
```diff
@@ -40,4 +40,4 @@ gui-js/documentation/*
 gui-js/dist/*
 gui-js/.nx/*
 gui-js/.angular/*
-doc/minsky/*
+doc/minsky/*
```
[the remark this answers is not in the document]
Comment (wdednam) [reply .gitignore:43]: I think it was Nano that did that, so I won't use it again. For coding and git operations, I use VS Code with the WSL extension.

**Pull request merged**